### PR TITLE
MCUmgr additions for DirectXIP with revert.

### PR DESCRIPTION
**Pull request by de-nordic**
Fromtrees up to a horizon.

### CODE REVIEW

**Comment by nordicjm** [comment]
`a43107299f6d4217f6730ccccf6a0e771357c3ca` should probably be here too

**Review by nordicjm — APPROVED** [review]

**Comment by de-nordic** [comment]
> `a43107299f6d4217f6730ccccf6a0e771357c3ca` should probably be here too

Where is that thing, can not find it on main.

**Comment by nordicjm** [comment]
```
commit a43107299f6d4217f6730ccccf6a0e771357c3ca
Author: Dominik Ermel <dominik.ermel@nordicsemi.no>
Date:   Mon Sep 25 13:47:32 2023 +0000

    doc/release-notes: Add info on downgrade prevention Kconfig
    
    Adds note on addition of
    MCUBOOT_BOOTLOADER_MODE_WITH_DOWNGRADE_PREVENTION
    to MCUboot Kconfig module.
    
    Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>
```

**Comment by nordicjm** [comment]
https://github.com/zephyrproject-rtos/zephyr/commit/a43107299f6d4217f6730ccccf6a0e771357c3ca

**Comment by de-nordic** [comment]
> [zephyrproject-rtos/zephyr@a431072](https://github.com/zephyrproject-rtos/zephyr/commit/a43107299f6d4217f6730ccccf6a0e771357c3ca)

Heh, I have skipped the release notes.

**Comment by de-nordic** [comment]
Added it now

**Review by kapi-no — APPROVED** [review]

File: modules/Kconfig.mcuboot [hunk 1]
```diff
@@ -154,30 +156,79 @@ config MCUBOOT_BOOTLOADER_MODE_SINGLE_APP
 
 config MCUBOOT_BOOTLOADER_MODE_SWAP_WITHOUT_SCRATCH
 	bool "MCUboot has been configured for swap without scratch operation"
+	select MCUBOOT_BOOTLOADER_MODE_HAS_NO_DOWNGRADE
 	help
 	  MCUboot expects slot0_partition and slot1_partition to be present
 	  in DT and application will boot from slot0_partition.
+	  MCUBOOT_BOOTLOADER_NO_DOWNGRADE should also be selected
+	  if MCUboot has been built with MCUBOOT_DOWNGRADE_PREVENTION.
```
Comment (glarsennordic) [inline modules/Kconfig.mcuboot:164]: > MCUBOOT_BOOTLOADER_NO_DOWNGRADE should also be selected if MCUboot has been built with MCUBOOT_DOWNGRADE_PREVENTION.

What is the motivation for a comment about `MCUBOOT_DOWNGRADE_PREVENTION.` being in the `MCUBOOT_BOOTLOADER_MODE_SWAP_WITHOUT_SCRATCH` help-text?

I don't see it as a problem or anything, but it makes me wonder if maybe there is a copy and paste error that explains it?

**Review by glarsennordic — COMMENTED** [review]
I don't think I'm actually a code owner of any of this, so I have not reviewed super closely. I also really don't understand XIP at all. But everything here looks reasonable at first glance.

Comment only since this isn't my purview

File: modules/Kconfig.mcuboot (hunk 1 above)
Comment (de-nordic) [reply modules/Kconfig.mcuboot:164]: Downgrade prevention is optional feature for swap algorithms that will prevent booting new image with lower version. 
DirectXIP (not XIP) always has downgrade prevention present.

**Review by mkapala-nordic — APPROVED** [review]

**Pull request merged**